kubernetes/istio_rules_service.go

```diff
@@ -0,0 +1,167 @@
+package kubernetes
+
+import (
+	"fmt"
+	"strings"
+)
+
+// GetIstioRules returns a list of mixer rules for a given namespace.
+func (in *IstioClient) GetIstioRules(namespace string) (*IstioRules, error) {
+	result, err := in.istio.Get().Namespace(namespace).Resource(rules).Do().Get()
+	if err != nil {
+		return nil, err
+	}
+	ruleList, ok := result.(*ruleList)
+	if !ok {
+		return nil, fmt.Errorf("%s doesn't return a rules list", namespace)
+	}
+
+	istioRules := IstioRules{}
+	istioRules.Rules = make([]IstioObject, 0)
+	for _, rule := range ruleList.Items {
+		istioRules.Rules = append(istioRules.Rules, rule.DeepCopyIstioObject())
+	}
+
+	return &istioRules, nil
+}
+
+// GetIstioRuleDetails returns the handlers and instances details for a given mixer rule.
+// On this version, the following handlers and instances are supported:
+// 		- listchecker
+// 		- listentry
+//		- denier
+// 		- checknothing
+func (in *IstioClient) GetIstioRuleDetails(namespace string, istiorule string) (*IstioRuleDetails, error) {
+	result, err := in.istio.Get().Namespace(namespace).Resource(rules).SubResource(istiorule).Do().Get()
+	if err != nil {
+		return nil, err
+	}
+	mRule, ok := result.(*rule)
+	if !ok {
+		return nil, fmt.Errorf("%s/%s doesn't return a Rule", namespace, istiorule)
+	}
+
+	istioRuleDetails := IstioRuleDetails{}
+	istioRuleDetails.Rule = mRule.DeepCopyIstioObject()
+
+	actions, ok := istioRuleDetails.Rule.GetSpec()["actions"].(actionsType)
+	if !ok {
+		return nil, fmt.Errorf("%s/%s doesn't return an Action", namespace, istiorule)
+	}
+
+	istioRuleDetails.Actions = make([]*IstioRuleAction, 0)
+	for _, rawAction := range actions {
+		action, ok := rawAction.(actionType)
+		if !ok {
+			return nil, fmt.Errorf("%s doesn't follow a map format", action)
+		}
+		actionDetails, err := in.getActionDetails(namespace, istiorule, action)
+		if err != nil {
+			return nil, err
+		}
+		istioRuleDetails.Actions = append(istioRuleDetails.Actions, actionDetails)
+	}
+
+	return &istioRuleDetails, nil
+}
+
+func (in *IstioClient) getActionDetails(namespace string, istiorule string, action actionType) (*IstioRuleAction, error) {
```

Comment:
IMO this method should be a composition of calls to small self-contained methods.